x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,209 @@
+         - metricset: diskio
+           dataset: system.diskio
+         - metricset: load
+           dataset: system.load
+         - metricset: memory
+           dataset: system.memory
+         - metricset: process
+           dataset: system.process
+           processes: ["firefox*"]
+         - metricset: process_summary
+           dataset: system.process_summary
+         - metricset: uptime
+           dataset: system.uptime
+         - metricset: socket_summary
+           dataset: system.socket_summary
+         - metricset: filesystem
+           dataset: system.filesystem
+         - metricset: raid
+           dataset: system.raid
+         - metricset: socket
+           dataset: system.socket
+         - metricset: service
+           dataset: system.service
+         - metricset: fsstat
+           dataset: system.fsstat
+         - metricset: foo
+           dataset: system.foo
+
+#################################################################################################
+### suggestion 1
+ - id: myendpoint-x1
+   title: Endpoint configuration
+   package:
+     name: endpoint
+     version: xxx
+   inputs:
+     - type: endpoint # Reserved key word
+       streams:
+         - type: malware
+           detect: true
+           prevent: false
+           notify_user: false
+           threshold: recommended
+           platform: windows
+
+        - type: eventing
+          api: true
+          clr: false
+          dll_and_driver_load: false
+          dns: true
+          file: false
+          platform: windows
+
+       - type: malware
+         detect: true
+         prevent: false
+         notify_user: false
+         threshold: recommended
+         platform: mac
+
+      - type: eventing
+        api: true
+        clr: false
+        dll_and_driver_load: false
+        dns: true
+        file: false
+        platform: mac
+
+       - type: malware
+         detect: true
+         prevent: false
+         notify_user: false
+         threshold: recommended
+         platform: linux
+
+      - type: eventing
+        api: true
+        clr: false
+        dll_and_driver_load: false
+        dns: true
+        file: false
+        platform: linux
+
+#################################################################################################
+### suggestion 2
+ - id: myendpoint-1
+   title: Endpoint configuration
+   package:
+     name: epm/endpoint # This establish the link with the package and will allow to link it to endpoint app.
+     version: xxx
+   inputs:
+     - type: endpoint # Reserved key word
+       windows:
+            eventing:
+                api: true
+                clr: false
+                dll_and_driver_load: false
+                dns: true
+                 ...
+                file: false
+            malware:
+                detect: true
+                prevent: false
+                notify_user: false
+                threshold: recommended
```

Comment:
I know this is an example, but some of these could change.  For instance we've talked about dropping `threshold`, is this still the case @ferullo @stevewritescode

Reply:
From an agent / fleet prespective, everything under input except `type` we don't know about and just forward. Any changes on this will not affect the agent itself.